crates/autopilot/src/domain/settlement/transaction/mod.rs

```diff
@@ -156,6 +190,28 @@ fn is_settlement_trace(trace: &eth::CallFrame, settlement_contract: eth::Address
         && trace.input.0.starts_with(&*SETTLE_FUNCTION_SELECTOR)
 }
 
+async fn find_solver_address<T: Authenticator>(
+    authenticator: &T,
+    path: Vec<&eth::CallFrame>,
+) -> Result<Option<eth::Address>, Error> {
+    let valid_solvers: Vec<Option<eth::Address>> =
+        // The RPC calls to check each address can be done in parallel as they are cheap
+        futures::future::join_all(path.iter().map(|call| async {
+            match authenticator.is_valid_solver(call.from).await {
```

Comment:
It's technically possible that some time passes between the tx happening and us indexing it. If the tx was malicious and the solver got deny listed by the circuit breaker because of it we wouldn't find an eligible caller in the callstack here.
To avoid this case the underlying `authenticator::is_solver()` call needs to happen on the same block the transaction happened. Meaning the `Authenticator` trait also needs to take the block number as a second argument.